### DISCUSSION TITLE
html option not usable when using setRemoteInstance

### DISCUSSION THREAD

**Discussion by spaceemotion** [spatie/browsershot#436] [Q&A]
I am trying to render a custom HTML page via the `html` option. However, due to the set up of my project, I have to use a remote instance instead of a local installation (via `setRemoteInstance`).

Since the HTML file is being generated on a different server than the one creating the PDF, puppeteer is throwing the following error:

> Error: net::ERR_FILE_NOT_FOUND at file:///tmp/942110971-0956970001603573522/index.html at navigate 

I know I can generate a URL that can be opened instead, but that requires loading the data twice (apart from me having difficulties with this approach as well, just running into timeouts right now).

---

Alternative: Render custom HTML via `setContent` (around line 172 in browser.js):

```php
$browsershot->setOption('html', view('xxxx')->render())
```

```js
if (request.options.html) {
  await page.setContent(request.options.html);
} else {
  await page.goto(request.url, requestOptions);
}
```

While this is a quick proof of concept that works, there is a limit to the payload size. It seems to cap out at around 120k characters. Probably not enough for bigger files.

Thankfully the following call eliminates this issue:

```php
$browsershot->writeOptionsToFile()
```

**Comment by apatkinson** [comment]
I am also getting this problem as described... bumping for a resolution.
